flang/lib/Lower/OpenACC.cpp

```diff
@@ -107,6 +107,18 @@ static Op createRegionOp(Fortran::lower::FirOpBuilder &builder,
   return op;
 }
 
+template <typename Op>
+static Op createSimpleOp(Fortran::lower::FirOpBuilder &builder,
+                         mlir::Location loc,
+                         const SmallVectorImpl<Value> &operands,
+                         const SmallVectorImpl<int32_t> &operandSegments) {
+  llvm::ArrayRef<mlir::Type> argTy;
+  Op op = builder.create<Op>(loc, argTy, operands);
```

Comment:
Same code as previous PR?

Reply:
Yes same code. I separated the PR in similar chunk as it will be upstream. So the templated function is the same.